### PR TITLE
Feature/vega 5186

### PR DESCRIPTION
**Pull request by ndrzhn**
## Problem statement

https://artcpt.atlassian.net/browse/VEGA-5186

## Solution details

 Интегрирован подбор по данным с бэкендом

## Type

- [x] Feature
- [ ] Bug Fix
- [ ] Refactoring
- [ ] Test
- [ ] Other

## Associated issues

https://artcpt.atlassian.net/browse/VEGA-4541

## Quality control

- [x] PR: Based on a correct branch
- [x] PR: Head branch is rebased on the base branch
- [x] PR: Assignee chosen and necessary labels are set
- [x] PR: Current form is filled out (where it makes sense)
- [x] PR: Diff checked for irrelevant changes
- [x] PR: Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specs
- [x] PR: Checked for spelling, grammar and typos
- [x] JS: There's no errors/warnings in the browser dev console
- [x] Layout: Tested with various content amounts
- [ ] Docs: All the important changes and features are described
- [ ] Tests: Configuration files are checked at test repositories
- [ ] Tests: New unit tests developed
- [ ] Tests: New E2E tests developed
- [x] Tests: Existing unit tests passed successfully
- [x] Tests: Existing E2E tests passed successfully
- [x] Tests: Manually tested on personal instance

## Notable statements
- [ ] Affects configuration files
- [ ] Brings new packages or updates existing
- [ ] Opened follow-up tasks to resolve

Screenshot 2022-04-04 at 18 51 16
http://jenkins-remote-ci-gpn-vega-builder.vega2.org:8081/job/STAND-TOOLBOX/job/scenario-tests-on-demand/2190/testReport/

### CODE REVIEW

File: src/components/DistributionSelectionContent/DistributionSelectionContent.tsx [hunk 1]
```diff
@@ -87,19 +118,21 @@ const DistributionSelectionContent: FC<DistributionSelectionContentProps> = ({
...
-        columns={table.columns}
-        rows={table.rows}
-        size="s"
-        borderBetweenColumns
-        borderBetweenRows
-      />
-    );
+    return tableIsLoading ?
+    <Loader size="m" data-testid="custom-distribution-table-loader" /> : (
+      <DistributionStatisticsComponent statistic={table} />
```
Comment (Alexandr1111) [inline src/components/DistributionSelectionContent/DistributionSelectionContent.tsx:123]: statistic**S**

Comment (ndrzhn) [reply src/components/DistributionSelectionContent/DistributionSelectionContent.tsx:123]: точно, спасибо, поправила

File: src/components/DistributionSelectionContent/DistributionSelectionContent.tsx [hunk 2]
```diff
@@ -63,6 +68,32 @@ const DistributionSelectionContent: FC<DistributionSelectionContentProps> = ({
     }
   }, [currentDistribution, onChange]);
 
+  useEffect(() => {
+    dispatch(CustomDistributionAction.loadTable({
+      conception: conceptionName || '', 
+      domainObjectId: (selectedCell.row.key?.value as string) || '',
+      attributeCode: selectedCell.column.key || '',
+      distributionType: currentDistribution?.type || null,
+    }));
```
[the remark this answers is not in the document]
Comment (ndrzhn) [reply src/components/DistributionSelectionContent/DistributionSelectionContent.tsx:77]: переименовала

File: src/components/DistributionSelectionContent/DistributionSelectionStatistics/DistributionSelectionStatistics.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+interface Props {
+  statistic: DistributionStatistic;
+}
+
+const getMappedColumn = (
+  header: DistributionStatisticHeaders,
+  isLeft = true,
+): TableColumn<TableRow> => {
+  const title = header.name.toLowerCase();
```
[the remark this answers is not in the document]
Comment (ndrzhn) [reply src/components/DistributionSelectionContent/DistributionSelectionStatistics/DistributionSelectionStatistics.tsx:21]: поправила

File: src/components/DistributionSettings/UserDistributionSettingsFormFields/UploadDistributionFileForm.tsx [hunk 4]
```diff
@@ -53,11 +56,16 @@ export const UploadDistributionFileForm: React.FC<
...
+        distributionType === DistributionTypes.CustomSample ?
+          await projectService.loadSampledAttributeValue(
+            attribute,
+            rowId,
+            loadedFile.vid,
+          ) : await projectService.loadSampledAttributeValueWithSelection(
+            attribute,
+            rowId,
+            loadedFile.vid,
+          );
```
[the remark this answers is not in the document]
Comment (ndrzhn) [reply src/components/DistributionSettings/UserDistributionSettingsFormFields/UploadDistributionFileForm.tsx:71]: вынесла

File: src/components/DistributionSettings/UserDistributionSettingsFormFields/UserDistributionSettingsFormFields.tsx [hunk 5]
```diff
@@ -60,6 +61,15 @@ export const UserDistributionSettingsFormFields: React.FC<
       ...formData,
       ...distribution,
     });
+
+    if (
+      distribution.sourceVid && (
+        distribution.distributionType === DistributionTypes.CustomDataSelection ||
+        formData.distributionType === DistributionTypes.CustomDataSelection
```
[the remark this answers is not in the document]
Comment (ndrzhn) [reply src/components/DistributionSettings/UserDistributionSettingsFormFields/UserDistributionSettingsFormFields.tsx:65]: вынесла

File: src/components/DistributionSelectionContent/DistributionSelectionContent.tsx [hunk 6]
```diff
@@ -63,6 +68,32 @@ const DistributionSelectionContent: FC<DistributionSelectionContentProps> = ({
...
+      hideDistribution: hideInputDistribution,
+    }));
+  }, [
+    conceptionName,
+    currentDistribution,
+    dispatch,
+    hideInputDistribution,
+    selectedCell.column.key,
+    selectedCell.row.key?.value,
+  ]);
```
[the remark this answers is not in the document]
Comment (ndrzhn) [reply src/components/DistributionSelectionContent/DistributionSelectionContent.tsx:95]: оставила тут из-за того, что не все данные для пэйлоада можно взять из стора

File: src/store/customDistribution/customDistributionEpics.ts [hunk 7]
```diff
@@ -68,18 +64,11 @@ const handleGetTable: Epic<
   StoreDependencies
 > = (action$, state$, { dispatch }) =>
   action$.pipe(
-    ofAction(
-      CustomDistributionAction.loadHistogram,
-      CustomDistributionAction.setHideInputDistribution,
-      CustomDistributionAction.setCurrentDistribution,
-    ),
+    ofAction(CustomDistributionAction.loadTable),
```
[the remark this answers is not in the document]
Comment (ndrzhn) [reply src/store/customDistribution/customDistributionEpics.ts:67]: аналогично: оставила сверху из-за отсутствия некоторых нужных данных в сторе

File: src/components/DistributionSelectionContent/DistributionSelectionContent.tsx [hunk 8]
```diff
@@ -22,27 +19,31 @@ import {
...
+  conceptionName?: string;
   onChange?: (item: DistributionItem | null) => void;
   closeModal: () => void;
   handleSetDistribution: (item: DistributionItem | null) => void;
   isLoading: boolean;
 }
 
 const DistributionSelectionContent: FC<DistributionSelectionContentProps> = ({
+  selectedCell,
+  conceptionName,
```
[the remark this answers is not in the document]
Comment (ndrzhn) [reply src/components/DistributionSelectionContent/DistributionSelectionContent.tsx:39]: да, поменяла

File: src/components/DistributionSelectionContent/DistributionSelectionContent.tsx [hunk 9]
```diff
@@ -53,7 +54,11 @@ const DistributionSelectionContent: FC<DistributionSelectionContentProps> = ({
   const hideInputDistribution = useSelector(hideInputDistributionSelector);
 
   useMount(() => {
-    dispatch(CustomDistributionAction.loadDistributions());
+    dispatch(CustomDistributionAction.loadDistributions({
+      conception: conceptionName || '', 
+      domainObjectId: (selectedCell.row.key?.value as string) || '',
```
[the remark this answers is not in the document]
Comment (ndrzhn) [reply src/components/DistributionSelectionContent/DistributionSelectionContent.tsx:59]: selectedCell нет в сторе, так что оставила так

File: src/components/DistributionSelectionContent/DistributionSelectionContent.tsx [hunk 10]
```diff
@@ -63,6 +68,32 @@ const DistributionSelectionContent: FC<DistributionSelectionContentProps> = ({
...
+  }, [conceptionName, currentDistribution, dispatch, selectedCell.column.key, selectedCell.row.key?.value]);
+
+  useEffect(() => {
+    dispatch(CustomDistributionAction.loadHistogram({
+      conception: conceptionName || '', 
+      domainObjectId: (selectedCell.row.key?.value as string) || '',
+      attributeCode: selectedCell.column.key || '',
+      distributionType: currentDistribution?.type || null,
+      hideDistribution: hideInputDistribution,
+    }));
```
[the remark this answers is not in the document]
Comment (ndrzhn) [reply src/components/DistributionSelectionContent/DistributionSelectionContent.tsx:87]: это и то, и другое

**Review by SergeyKatugin — APPROVED** [review]

**Pull request merged**